numpy/random/tests/test_random.py

```diff
@@ -352,40 +356,41 @@ def test_bytes(self):
         np.testing.assert_equal(actual, desired)
 
     def test_shuffle(self):
```

Comment:
You can set the `description` attribute of the yielded test to improve the output, e.g.
```
import numpy as np
from numpy.testing import assert_allclose

def test_foo():

    def allclose(description):
        f = lambda x, y: assert_allclose(x, y)
        f.description = description
        return f

    for i in range(100):
        A = np.arange(100)
        if i != 50:
            B = A
        else:
            B = np.arange(200)
        yield allclose("test_foo:test #{}".format(i)), A, B

def test_bar():
    for i in range(100):
        A = np.arange(100)
        if i != 50:
            B = A
        else:
            B = np.arange(200)
        assert_allclose(A, B)
```
gives, IMO, a nice output:
```
..................................................F.................................................F
======================================================================
FAIL: test_foo:test #50
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/tmp/test_foo.py", line 7, in <lambda>
    f = lambda x, y: assert_allclose(x, y)
  File "/usr/lib/python3.5/site-packages/numpy/testing/utils.py", line 1347, in assert_allclose
    verbose=verbose, header=header)
  File "/usr/lib/python3.5/site-packages/numpy/testing/utils.py", line 663, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

(shapes (100,), (200,) mismatch)
 x: array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50,...
 y: array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,...

======================================================================
FAIL: test_foo.test_bar
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/tmp/test_foo.py", line 26, in test_bar
    assert_allclose(A, B)
  File "/usr/lib/python3.5/site-packages/numpy/testing/utils.py", line 1347, in assert_allclose
    verbose=verbose, header=header)
  File "/usr/lib/python3.5/site-packages/numpy/testing/utils.py", line 663, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

(shapes (100,), (200,) mismatch)
 x: array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50,...
 y: array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,...

----------------------------------------------------------------------
Ran 101 tests in 0.056s

FAILED (failures=2)
```